python/cudf/cudf/_lib/cpp/io/orc_metadata.pxd

```diff
@@ -1,19 +1,77 @@
-# Copyright (c) 2020-2021, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
+from libc.stdint cimport int32_t, int64_t, uint32_t, uint64_t
+from libcpp cimport bool
+from libcpp.optional cimport optional
 from libcpp.string cimport string
 from libcpp.vector cimport vector
 
 cimport cudf._lib.cpp.io.types as cudf_io_types
+from cudf._lib.variant cimport monostate, variant
 
 
 cdef extern from "cudf/io/orc_metadata.hpp" \
         namespace "cudf::io" nogil:
 
-    cdef cppclass raw_orc_statistics:
+    cdef cppclass no_statistics(monostate):
+        pass
+
+    cdef cppclass minmax_statistics[T]:
+        optional[T] minimum
+        optional[T] maximum
+
+    cdef cppclass sum_statistics[T]:
+        optional[T] sum
+
+    cdef cppclass integer_statistics(
+        minmax_statistics[int64_t], sum_statistics[int64_t]
+    ):
+        pass
+
+    cdef cppclass double_statistics(
+        minmax_statistics[double], sum_statistics[double]
+    ):
+        pass
+
+    cdef cppclass string_statistics(
+        minmax_statistics[string], sum_statistics[int64_t]
+    ):
+        pass
+
+    cdef cppclass bucket_statistics:
+        vector[int64_t] count
+
+    cdef cppclass decimal_statistics(
+        minmax_statistics[string], sum_statistics[string]
+    ):
+        pass
+
+    cdef cppclass date_statistics(minmax_statistics[int32_t]):
+        pass
+
+    cdef cppclass binary_statistics(sum_statistics[int64_t]):
+        pass
```

Comment:
For the things that are type aliases in libcudf (`using` declarations) a more faithful representation in Cython is probably a ctypedef, e.g. `ctypedef minmax_statistics[int32_t] date statistics`.

Reply:
Good idea. I did this.